netket/experimental/driver/tdvp.py

```diff
@@ -39,32 +40,43 @@
 
 
 @dispatch
-def dwdt(state, driver, t, w, *, stage=None):
+def make_odefun(state, driver):
     # pylint: disable=unused-argument
-    raise NotImplementedError(f"dwdt not implemented for {type(state)}")
+    raise NotImplementedError(f"make_odefun not implemented for {type(state)}")
 
 
 @dispatch
-def dwdt(state: MCState, driver, t, w, *, stage: int = None):
-    # pylint: disable=protected-access
-    state.parameters = w
-    state.reset()
+def make_odefun(state: MCState, driver):
+    def odefun(args: dict):
+        # pylint: disable=protected-access
```

Comment:
this forces recompilation every time a new ode fun is constructed, even if identical.
Maybe you should return an HashablePartial..?

Reply:
Returning `HashablePartial` did not prevent recompilation when creating a new driver for the same system when I tested it. Given that this is not too much of an additional cost at the moment and repeatedly recreating drivers is not a common use case, I think we can accept the recompilation for this PR and improve the situation later.